magenta/models/rl_tuner/rl_tuner_ops.py

```diff
@@ -0,0 +1,302 @@
+"""Helper functions to support the RLTuner and NoteRNNLoader classes."""
+
+import os
+import random
+
+from magenta.common import tf_lib
+
+import numpy as np
+import tensorflow as tf
+
+LSTM_STATE_NAME = 'lstm'
+
+# Number of output note classes. This is a property of the dataset.
+NUM_CLASSES = 38
+
+# Default batch size.
+BATCH_SIZE = 128
+
+# Music-related constants.
+INITIAL_MIDI_VALUE = 48
+NUM_SPECIAL_EVENTS = 2
+MIN_NOTE = 48  # Inclusive
+MAX_NOTE = 84  # Exclusive
+TRANSPOSE_TO_KEY = 0  # C Major
+DEFAULT_BPM = 80.0
+
+# Music theory constants used in defining reward functions.
+# Note that action 2 = midi note 48.
+C_MAJOR_SCALE = [2, 4, 6, 7, 9, 11, 13, 14, 16, 18, 19, 21, 23, 25, 26]
+C_MAJOR_KEY = [0, 1, 2, 4, 6, 7, 9, 11, 13, 14, 16, 18, 19, 21, 23, 25, 26, 28,
+               30, 31, 33, 35, 37]
+C_MAJOR_TONIC = 14
+A_MINOR_TONIC = 23
+
+# The number of half-steps in musical intervals, in order of dissonance
+OCTAVE = 12
+FIFTH = 7
+THIRD = 4
+SIXTH = 9
+SECOND = 2
+FOURTH = 5
+SEVENTH = 11
+HALFSTEP = 1
+
+# Special intervals that have unique rewards
+REST_INTERVAL = -1
+HOLD_INTERVAL = -1.5
+REST_INTERVAL_AFTER_THIRD_OR_FIFTH = -2
+HOLD_INTERVAL_AFTER_THIRD_OR_FIFTH = -2.5
+IN_KEY_THIRD = -3
+IN_KEY_FIFTH = -5
+
+# Indicate melody direction
+ASCENDING = 1
+DESCENDING = -1
+
+# Indicate whether a melodic leap has been resolved or if another leap was made
+LEAP_RESOLVED = 1
+LEAP_DOUBLED = -1
+
+
+def default_hparams():
+  """Generates the hparams used to train note rnn used in paper."""
+  return tf_lib.HParams(use_dynamic_rnn=True,
+                        batch_size=BATCH_SIZE,
+                        lr=0.0002,
+                        l2_reg=2.5e-5,
+                        clip_norm=5,
+                        initial_learning_rate=0.5,
+                        decay_steps=1000,
+                        decay_rate=0.85,
+                        rnn_layer_sizes=[100],
+                        skip_first_n_losses=32,
+                        one_hot_length=NUM_CLASSES,
+                        exponentially_decay_learning_rate=True)
+
+def large_model_hparams():
+  """Generates the default hparams used to train a large note rnn."""
+  return tf_lib.HParams(use_dynamic_rnn=True,
+                        batch_size=BATCH_SIZE,
+                        lr=0.0002,
+                        l2_reg=2.5e-5,
+                        clip_norm=5,
+                        initial_learning_rate=0.5,
+                        decay_steps=1000,
+                        decay_rate=0.85,
+                        rnn_layer_sizes=[2500],
+                        skip_first_n_losses=8,
+                        one_hot_length=NUM_CLASSES,
+                        exponentially_decay_learning_rate=True)
+
+def basic_rnn_hparams():
+  """Generates the hparams used to train a basic_rnn."""
```

Comment:
Are these just the hparams that were used specifically for the basic_rnn.mag file? If so, we should indicate that in the docs and add a TODO comment to be able to restore a basic_rnn bundle with any kind of metagraph.